[google_maps_flutter] iOS - undefined symbols for architecture

I've been really struggling to get xcode to a build a project that uses google maps flutter. 

I get the following errors:

```
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_GMSMarker", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapMarkerController.o)
  "_OBJC_CLASS_$_GMSMutablePath", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapPolylineController.o)
  "_OBJC_CLASS_$_GMSCameraPosition", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapController.o)
  "_OBJC_CLASS_$_GMSCameraUpdate", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapController.o)
  "_OBJC_CLASS_$_GMSCoordinateBounds", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapController.o)
  "_OBJC_CLASS_$_GMSMapView", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapController.o)
  "_OBJC_CLASS_$_GMSPolyline", referenced from:
      objc-class-ref in google_maps_flutter(GoogleMapPolylineController.o)
  "_kGMSMinZoomLevel", referenced from:
      _InterpretMapOptions in google_maps_flutter(GoogleMapController.o)
  "_kGMSMaxZoomLevel", referenced from:
      _InterpretMapOptions in google_maps_flutter(GoogleMapController.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I'm using a Flutter module embedded within an app. This app already uses Google Maps.

When `s.weak_framework = 'GoogleMaps'` is added to `google_maps_flutter.podspec` it gets further, but still fails:
```
Ld /Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Products/Debug-iphonesimulator/FlutterPluginRegistrant/FlutterPluginRegistrant.framework/FlutterPluginRegistrant normal x86_64 (in target 'FlutterPluginRegistrant' from project 'Pods')
    cd /Users/m/Projects/flutter-swift-test-31104/SampleiOSFlutter/Pods
    /Users/m/Applications/Xcode-beta6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target x86_64-apple-ios8.0-simulator -dynamiclib -isysroot /Users/m/Applications/Xcode-beta6.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.0.sdk -L/Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Products/Debug-iphonesimulator/FlutterPluginRegistrant -F/Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Products/Debug-iphonesimulator/FlutterPluginRegistrant -F/Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Products/Debug-iphonesimulator/google_maps_flutter -F/Users/m/Projects/flutter-swift-test-31104/SampleiOSFlutter/Pods/../../FlutterModule/.ios/Flutter/engine -F/Users/m/Projects/flutter-swift-test-31104/SampleiOSFlutter/Pods/GoogleMaps/Base/Frameworks -F/Users/m/Projects/flutter-swift-test-31104/SampleiOSFlutter/Pods/GoogleMaps/Maps/Frameworks -filelist /Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FlutterPluginRegistrant.build/Objects-normal/x86_64/FlutterPluginRegistrant.LinkFileList -install_name @rpath/FlutterPluginRegistrant.framework/FlutterPluginRegistrant -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FlutterPluginRegistrant.build/Objects-normal/x86_64/FlutterPluginRegistrant_lto.o -Xlinker -export_dynamic -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -weak_framework GoogleMaps -framework Foundation -framework google_maps_flutter -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FlutterPluginRegistrant.build/Objects-normal/x86_64/FlutterPluginRegistrant_dependency_info.dat -o /Users/m/Library/Developer/Xcode/DerivedData/SampleiOSFlutter-dynfdplugzidvkdmfalgbckdklna/Build/Products/Debug-iphonesimulator/FlutterPluginRegistrant/FlutterPluginRegistrant.framework/FlutterPluginRegistrant

Undefined symbols for architecture x86_64:
  "_GMSx_GMTTDDistance_Units_EnumDescriptor", referenced from:
      _descriptor.fields in GoogleMaps
  "_OBJC_METACLASS_$_GMSx_IVRenderLayer", referenced from:
      _OBJC_METACLASS_$_GMSRocketIVRenderLayer in GoogleMaps
  "_OBJC_METACLASS_$_GMSx_IVCameraLayer", referenced from:
      _OBJC_METACLASS_$_GMSRocketIVCameraLayer in GoogleMaps
  "_OBJC_METACLASS_$_GMSTileServices", referenced from:
      _OBJC_METACLASS_$_GMSVectorTileServices in GoogleMaps
  "_OBJC_CLASS_$_GMSTileServices", referenced from:
      _OBJC_CLASS_$_GMSVectorTileServices in GoogleMaps
  "_OBJC_METACLASS_$_GMSBaseResources", referenced from:
      _OBJC_METACLASS_$_GMSMapsResources in GoogleMaps
  "_OBJC_CLASS_$_GMSBaseResources", referenced from:
      _OBJC_CLASS_$_GMSMapsResources in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GPBEnumDescriptor", referenced from:
      objc-class-ref in GoogleMaps
  "gmscore::utils::JavaDataOutput::WriteUTF(NSString*)", referenced from:
      -[GMSDASHConnection buildServerPayloadForRequests:] in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMTrafficIncidentsResponseProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMTactilePhotoResponseProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GPBCodedInputStream", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GPBCodedOutputStream", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMClientPropertiesResponseProto", referenced from:
      objc-class-ref in GoogleMaps
  "std::length_error::~length_error()", referenced from:
      std::__1::__split_buffer<double, std::__1::allocator<double>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<double>&) in GoogleMaps
      std::__1::__split_buffer<CLLocationCoordinate2D, std::__1::allocator<CLLocationCoordinate2D>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<CLLocationCoordinate2D>&) in GoogleMaps
      std::__1::__split_buffer<GMSStrokeStyle* __strong, std::__1::allocator<GMSStrokeStyle* __strong>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<GMSStrokeStyle* __strong>&) in GoogleMaps
      std::__1::__split_buffer<GMSOffscreenEntityView* __strong, std::__1::allocator<GMSOffscreenEntityView* __strong>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<GMSOffscreenEntityView* __strong>&) in GoogleMaps
      std::__1::__split_buffer<UIView* __strong, std::__1::allocator<UIView* __strong>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<UIView* __strong>&) in GoogleMaps
      std::__1::__split_buffer<gmscore::base::reffed_ptr<gmscore::renderer::Behavior>, std::__1::allocator<gmscore::base::reffed_ptr<gmscore::renderer::Behavior> >&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<gmscore::base::reffed_ptr<gmscore::renderer::Behavior> >&) in GoogleMaps
      std::__1::__split_buffer<std::__1::vector<gmscore::model::Point2D, std::__1::allocator<gmscore::model::Point2D> >, std::__1::allocator<std::__1::vector<gmscore::model::Point2D, std::__1::allocator<gmscore::model::Point2D> > >&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<std::__1::vector<gmscore::model::Point2D, std::__1::allocator<gmscore::model::Point2D> > >&) in GoogleMaps
      ...
  "_OBJC_CLASS_$_GMSx_GMMClientInformation_PixelDensity", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GmmClearcutCountersDimensions", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSFloorPickerViewController", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GPSImageKey", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSTileDataCache", referenced from:
      objc-class-ref in GoogleMaps
  "_GMSCreateDomainForClass", referenced from:
      _GMSRequestPerformanceStatusKey in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMPStylerDescription_Parameter", referenced from:
      objc-class-ref in GoogleMaps
  "gmscore::renderer::Resource::UsingRef()", referenced from:
      vtable for (anonymous namespace)::RocketRenderTarget in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMPStylerDescription", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMMapTileRequestProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMParameterGroupProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMApiQuotaEventResponseProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMApiQuotaEventRequestProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_EIDClientEventIdMessage", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMUserEvent3", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSCoreServerResourceManager", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSStyleTableService", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_GMMMapTilePaintDescriptionProto", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSLicensesCache", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSResourceUtils", referenced from:
      objc-class-ref in GoogleMaps
  "_OBJC_CLASS_$_GMSx_PlaceId", referenced from:
      objc-class-ref in GoogleMaps
  "gmscore::vector::Camera::Camera(gmscore::base::reffed_ptr<gmscore::renderer::RenderTarget> const&, gmscore::vector::CameraPosition const&)", referenced from:
      -[GMSMapView projection] in GoogleMaps
      -[GMSMapView projectionForCamera:] in GoogleMaps
  "_OBJC_CLASS_$_GMSRenderTileProvider", referenced from:
      objc-class-ref in GoogleMaps
```
and on and on.  I don't know where those symbols are coming from, but they aren't public in the GoogleMapsCore/Base.  There's something missing there.

--- a/packages/flutter_tools/lib/src/plugins.dart
+++ b/packages/flutter_tools/lib/src/plugins.dart
@@ -383,6 +383,7 @@ Depends on all your plugins, and provides a function to register them.
   s.source_files =  "Classes", "Classes/**/*.{h,m}"
   s.source           = { :path => '.' }
   s.public_header_files = './Classes/**/*.h'
+  s.static_framework    = true
   s.dependency '{{framework}}'
   {{#plugins}}
   s.dependency '{{name}}'

